common/connections.py

```diff
@@ -5,79 +5,87 @@
 from re import search
 from typing import Dict, Callable, Any
 
+
 class Client:
-    def __init__(self, baseUrl:str):
+    def __init__(self, baseUrl: str):
         self.__baseUrl = baseUrl
         self.__ticket = None
         self.__session = None
-        
-    def __updateTicket(self, ticket:str=None):
-        if ticket != None:
+
+    def __updateTicket(self, ticket: str = None):
+        if ticket is not None:
             self.__ticket = ticket
 
-        if self.__ticket != None and self.__session != None:
+        if self.__ticket is not None and self.__session is not None:
             basicStr = 'BASIC ' + b64encode(bytes(self.__ticket, 'utf-8')).decode('utf-8')
             self.__session.headers.update({'Authorization': basicStr})
-            
-    def __processRequest(self, api:str, opUrl:str, version:str, params:Dict, headers:Dict, responseHandler:Callable[[Response], Any], errorHandler:Callable[[Response], Any], requestHandler:Callable[[str,Dict,Dict,Any], Any], payload:Any=None):
+
+    def __processRequest(self, api: str, opUrl: str, version: str, params: Dict, headers: Dict, responseHandler: Callable[[Response], Any], errorHandler: Callable[[Response], Any], requestHandler: Callable[[str, Dict, Dict, Any], Any], payload: Any = None, files: Dict = None):
         effectiveVersion = '1'
-        if version != None:
+        if version is not None:
             effectiveVersion = version
         effectiveUrl = self.__baseUrl + '/api/-default-/public/' + api + '/versions/' + effectiveVersion + '/' + opUrl
-        
+
         # OrderedDict to keep any ordering in provided data during mapping
         effectiveParams = OrderedDict()
         effectiveHeaders = OrderedDict()
-        
-        if params != None:
+        effectiveFiles = OrderedDict()
+
+        if params is not None:
             for k in params:
                 if isinstance(params[k], list):
                     if len(params[k]) > 0:
                         # Public v1 ReST API always expects concatenated multi-value params
                         effectiveParams[k] = ','.join(params[k])
                 else:
                     effectiveParams[k] = params[k]
-                
-        if headers != None:
+
+        if headers is not None:
             for k in headers:
                 effectiveHeaders[k] = headers[k]
-        
-        
-        if self.__session == None:
+
+        if files is not None:
+            for k in files:
+                effectiveFiles[k] = files[k]
+
+        if self.__session is None:
             self.__session = Session()
             self.__updateTicket()
-        
-        with requestHandler(effectiveUrl, effectiveParams, effectiveHeaders, payload) as response:
+
+        with requestHandler(effectiveUrl, effectiveParams, effectiveHeaders, effectiveFiles, payload) as response:
```

Comment:
I am wary of supporting multiple file uploads as the public API is restricted to only cm:content as a content property, and thus effectively only supports one content stream per document. None of the __doPost / __doJsonPost functions have been adapted to the changed signature, so only the new __doMultipartPost supports this.

Reply:
Don't think it's multiple file uploads... only one file is allowed...